### PR TITLE
Updated Tekton model to v0.11.0-rc2

### PR DESCRIPTION
**Pull request by rohanKanojia**
Kudos to @pradeepitm12 for helping me out on upgrading this.

### CODE REVIEW

File: extensions/tekton/generator/cmd/generate/generate.go [hunk 1]
```diff
@@ -16,43 +16,69 @@
...
+  V1alpha1TaskRunList                     v1alpha1.TaskRunList
+  V1alpha1ClusterTask                     v1alpha1.ClusterTask
+  V1alpha1ClusterTaskList                 v1alpha1.ClusterTaskList
+  V1alpha1Condition                       v1alpha1.Condition
+  V1alpha1ConditionList                   v1alpha1.ConditionList
+  V1alpha1PipelineTask                    v1alpha1.PipelineTask
+  V1alpha1PipelineTaskResource            v1alpha1.PipelineTaskResources
+  V1alpha1ResourceDeclaration             v1alpha1.ResourceDeclaration
+
+  //V1alpha2Pipeline                        v1alpha2.Pipeline
```
Comment (manusa) [inline extensions/tekton/generator/cmd/generate/generate.go]: Is there any reason to keep these commented-out lines?

**Comment by jeffmaury** [comment]
As I'm working on the same kind of changes (Triggers), I noticed that this PR will bring a breaking change as few classes have moved to a different packages and even some have their name changed. It is a issue ?

**Comment by rohanKanojia** [comment]
@jeffmaury : Hi, Could you please tell me classes which I moved to different package? I'm not so familiar with tekton so I could have done some mistake. Happy to modify my PR as per your suggestions :-) .

Or are you saying that you're also working on updating model and would like to update it yourself? I know you've already sent a patch to fix 2 issues on our tekton extensions so it would be better if you contribute more :smile: (whether it's code contribution or review)

**Comment by jeffmaury** [comment]
For what I recall (but that would need a need to diff comparison):
- Step has been moved to v1alpha2 package
- Template (pod) has been moved to a different pod package (was v1alpha1)
- Task params is now a list of v1alpha2.ParmSpec (was v1alpha1.Parm) see updated PipeLineCrudTest
- pipeline.v1alpha1.PipelineResource moved to resource.v1alpha1.PipelineResource

**Comment by jeffmaury** [comment]
Also these changes could be outlined with:
`mvn clean clirr:check`

**Comment by rohanKanojia** [comment]
@jeffmaury : I understand that this PR would break backward compatibility with previous versions, but I think it's due to breaking changes in Tekton and we would need to adapt accordingly? If you need release without this patch, we can cut a release without merging this? WDYT?

**Comment by jeffmaury** [comment]
I don't need that PR to be in for my purpose. However, I have a question regarding API versioning as Tekton is also v1beta1 now so what is the policy regarding API versions:
- expose all versions ports (not done by Knative for example)
- uses the latest version (but will break backward compatibility)

**Comment by rohanKanojia** [comment]
@jeffmaury : Sorry for the late reply, I just came back from PTO. In my opinion we should use the latest version since Knative and Tekton are moving very fast and it would be nice to stay updated to the latest versions.

**Comment by rohanKanojia** [comment]
Tekton folks have released another version: https://github.com/tektoncd/pipeline/releases . I will update this to that

**Comment by rohanKanojia** [comment]
@jeffmaury : Hi Jeff, I've updated model to `v0.11.0-rc2` and modified DSL to use `v1beta1` resources. Could you please take a look at this PR and see if this is the change you wanted?

**Comment by Fabian-K** [comment]
Hi,
great that you are already working on updating tekton to the latest version! I had a look at this pr and noticed that it has the same issue with `TaskResourceBinding` as we faced with the `Step`class.

Instead of having the two properties `paths` and `pipelineResourceBinding` it needs to have all properties of the `PipelineResourceBinding`(e.g. name, resourceRef, resourceSpec) and in addition paths.

For `Step` that was fixed by @jeffmaury with https://github.com/fabric8io/kubernetes-client/pull/2033.

As already mentioned, unfortunately I don´t know the generation logic. Is it possible to find a universal fix instead of workarounds for individual classes? Root cause seems to be the handling of embedded structs in go (see my comment in https://github.com/fabric8io/kubernetes-client/issues/1984#issuecomment-582415107).

**Comment by rohanKanojia** [comment]
@Fabian-K : Unfortunately, I'm also no so familiar with Golang either. But looking at code maybe `field.Anonymous && field.Type.Kind() == reflect.Struct` seems to be checking for Embedded Structs. I tried removing the check for length but that started breaking model generation process. In the end, I had to hardcode `TaskResourceBinding` like `Container` to get expected results.

https://github.com/fabric8io/kubernetes-client/blob/8321522e75d0e6f5b5ef0aadea3d82f07beff758/extensions/tekton/generator/pkg/schemagen/generate.go#L424

@iocanel : Do you have some idea on how to make handling of Embedded Structs a bit smarter?

File: extensions/tekton/generator/cmd/generate/generate.go [hunk 2]
```diff
@@ -16,43 +16,65 @@
...
 
 func main() {
 	packages := []schemagen.PackageDescriptor{
 		//"$ref": "#/definitions/github_com_tektoncd_pipeline_vendor_k8s_io_api_core_v1_Probe",
 		{"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1", "tekton.dev", "io.fabric8.tekton.pipeline.v1alpha1", "tekton_v1alpha1_"},
-		{"github.com/knative/pkg/apis", "v1", "io.fabric8.knative.v1", "knative_"},
+    {"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1", "tekton.dev", "io.fabric8.tekton.pipeline.v1beta1", "tekton_v1alpha2_"},
+    {"github.com/tektoncd/pipeline/pkg/apis/resource/v1alpha1", "tekton.dev", "io.fabric8.tekton.resource.v1alpha1", "tekton_resource_v1alpha1_"},
+    {"github.com/tektoncd/pipeline/pkg/apis/pipeline/pod", "tekton.dev", "io.fabric8.tekton.pipeline.pod", "tekton_pod_"},
+    {"github.com/knative/pkg/apis", "v1", "io.fabric8.knative.v1", "knative_"},
```
Comment (hrishin) [inline extensions/tekton/generator/cmd/generate/generate.go]: Why do we need?

Comment (rohanKanojia) [reply extensions/tekton/generator/cmd/generate/generate.go]: Umm, not sure. I didn't touch it while upgrading model. Let me try removing it.

Comment (rohanKanojia) [reply extensions/tekton/generator/cmd/generate/generate.go]: Looks like some structs in tekton model depend on these packages(one of these is this https://github.com/tektoncd/pipeline/blob/master/pkg/apis/pipeline/v1beta1/pipelinerun_types.go#L28 ), when I removed them I started getting errors.

Comment (hrishin) [reply extensions/tekton/generator/cmd/generate/generate.go]: ahh, no worries

**Comment by rohanKanojia** [comment]
[merge]

**Pull request merged**